tests/element_test_vectors.rs

```diff
@@ -196,26 +201,54 @@ trait ElementApi {
     fn read_group<F1, F2>(file_name: &str, value_assert: F1, group_assert: F2) -> IonResult<()>
     where
         // group index, value 1 index, value 1, value 2 index, value 2
-        F1: Fn(usize, usize, &Element, usize, &Element),
+        F1: Fn(usize, String, &Element, String, &Element),
         F2: Fn(&Vec<Element>, &Vec<Element>),
     {
         let group_lists = Self::read_file(file_name)?;
-        for (group_index, group_list) in group_lists.iter().enumerate() {
+        for (group_index, group_container) in group_lists.iter().enumerate() {
             // every grouping set is a list/sexp
             // look for the embedded annotation to parse/test as the underlying value
-            let is_embedded = group_list.annotations().contains("embedded_documents");
-            match (group_list.as_sequence(), is_embedded) {
-                (Some(group), true) => {
+            let is_embedded = group_container.annotations().contains("embedded_documents");
+            match (group_container.value(), is_embedded) {
+                (Value::List(group), true) | (Value::SExp(group), true) => {
                     Self::read_group_embedded(group, &group_assert)?;
                 }
-                (Some(group), false) => {
+                (Value::Struct(group), true) => {
+                    let sequence: Sequence = group
+                        .to_owned()
+                        .fields()
+                        .map(|(_, v)| v.to_owned())
+                        .collect();
```

Comment:
I think you can skip the `to_owned()` call on `group`.

Checking my understanding: this is discarding the field names (which are names for the different tests) because we don't really have a use for them at the moment, yes?